flow/connectors/clickhouse/clickhouse.go

```diff
@@ -77,24 +77,43 @@ func (c *ClickHouseConnector) ValidateCheck(ctx context.Context) error {
 	err := c.database.Exec(ctx, fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
 		id UInt64
 	) ENGINE = ReplacingMergeTree ORDER BY id;`,
-		validateDummyTableName+"_temp"))
+		validateDummyTableName))
 	if err != nil {
 		return fmt.Errorf("failed to create validation table %s: %w", validateDummyTableName, err)
 	}
+	defer func() {
+		// goroutine so API call doesn't hang
+		go func() {
+			// drop the table at any cost
+			for i := 1; i <= 5; i++ {
+				err = c.database.Exec(ctx, "DROP TABLE IF EXISTS "+validateDummyTableName)
```

Comment:
this will race on using connection if `c.database` is used after this method is called. Better to not use goroutine

Reply:
can be worked around since the connector only exists for the duration of the API call
but `TEMPORARY` would work better

Reply:
the better way would be to have caller responsible for returning status to user & on it's own it'd handle drop table before dropping connector